### PR TITLE
Updates @sap/approuter types to be able to add custom properties to the MiddlwareHandler request

### PR DESCRIPTION
**Pull request by nbblum**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

File: types/sap__approuter/index.d.ts [hunk 1]
```diff
@@ -60,23 +60,33 @@ declare namespace approuter {
          */
         backendTimeout?: (req: IncomingMessage, done: () => void) => void;
 
-        /**
-         * Callback for {@link afterRequestHandler}
-         *
-         * @param error - Error object in case of error
-         * @param incomingResponse - Modified incomingResponse
-         */
+        // In order to make middleware easier to write to be agnostic of the route, the req.url will be altered to remove the route part and the original will be available set to this value
```
Comment (biolauri) [inline types/sap__approuter/index.d.ts:63]:
```suggestion
        /** In order to make it easier to write middlewares being agnostic of the route, the `req.url` will be altered to remove the route part and the original URL will be set to this value */
```
I'd prefer JSDoc and improved some wording.

File: types/sap__approuter/index.d.ts [hunk 2]
```diff
@@ -60,23 +60,33 @@ declare namespace approuter {
...
+        originalUrl?: string;
+
+        // On each request, the application router executes registered middlewares in a certain order and the session is not available to all of them.
```
Comment (biolauri) [inline types/sap__approuter/index.d.ts:66]:
```suggestion
        /** Note: On each request, the application router executes registered middlewares in a certain order and the session is not available to all of them. */
```
Also JSDoc and some wording.

**Review by biolauri — APPROVED** [review]
LGTM 👍
Had just some optional minor suggestions.

**Review by biolauri — APPROVED** [review]
LGTM 👍

**Comment by nbblum** [comment]
Ready to merge

**Pull request merged**